macOS CI is failing due to Homebrew update

Example: https://github.com/pypa/pip/actions/runs/3278007787/jobs/5395928619

```
Run brew install bzr
Error: bazaar has been disabled because it is not supported. Check out `breezy` instead!
Error: Process completed with exit code 1.
```

We probably need some updates; maybe simply disable bazaar tests, maybe use breezy (which might need some updates in the VCS implementation).

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -120,7 +120,7 @@ jobs:
 
       - name: Install MacOS dependencies
         if: matrix.os == 'MacOS'
-        run: brew install bzr
+        run: brew install breezy
 
       - run: pip install nox 'virtualenv<20' 'setuptools != 60.6.0'
 

